app/docker/networks/edit/NetworkDetailsView.tsx

```diff
@@ -0,0 +1,130 @@
+import { useQueryClient } from 'react-query';
+
+import { useEnvironmentId } from '@/portainer/hooks/useEnvironmentId';
+import { PageHeader } from '@/portainer/components/PageHeader';
+import { confirmDeletionAsync } from '@/portainer/services/modal.service/confirm';
+import { AccessControlPanel } from '@/portainer/access-control/AccessControlPanel/AccessControlPanel';
+import { ResourceControlType } from '@/portainer/access-control/types';
+import { DockerContainer } from '@/docker/containers/types';
+
+import { useNetwork, useDeleteNetwork } from '../queries';
+import { isSystemNetwork } from '../network.helper';
+import { useContainers } from '../../containers/queries';
+import { DockerNetwork, NetworkContainer } from '../types';
+
+import { NetworkDetailsTable } from './NetworkDetailsTable';
+import { NetworkOptionsTable } from './NetworkOptionsTable';
+import { NetworkContainersTable } from './NetworkContainersTable';
+
+export function NetworkDetailsView() {
+  const router = useRouter();
+  const queryClient = useQueryClient();
+
+  const [filteredNetworkContainers, setFilteredNetworkContainers] = useState(
+    [] as NetworkContainer[]
+  );
+  const {
+    params: { id: networkId, nodeName },
+  } = useCurrentStateAndParams();
+  const environmentId = useEnvironmentId();
+
+  const networkQuery = useNetwork(environmentId, networkId);
+  const deleteNetworkMutation = useDeleteNetwork();
+  const filters = {
+    network: [networkId],
+  };
+  const containersQuery = useContainers(environmentId, filters);
+
+  useEffect(() => {
+    if (networkQuery.data && containersQuery.data) {
+      setFilteredNetworkContainers(
+        filterContainersInNetwork(networkQuery.data, containersQuery.data)
+      );
+    }
+  }, [networkQuery.data, containersQuery.data]);
+
+  return (
+    <>
+      <PageHeader
+        title="Network details"
+        breadcrumbs={[
+          { link: 'docker.networks', label: 'Networks' },
+          {
+            link: 'docker.networks.network',
+            label: networkQuery.data?.Name || '',
+          },
+        ]}
+      />
+      {networkQuery.data && (
+        <>
+          <NetworkDetailsTable
+            network={networkQuery.data}
+            onRemoveNetworkClicked={onRemoveNetworkClicked}
+          />
+
+          <AccessControlPanel
+            onUpdateSuccess={() =>
+              queryClient.invalidateQueries([
+                'environments',
+                environmentId,
+                'docker',
+                'networks',
+                networkId,
+              ])
+            }
+            resourceControl={networkQuery.data.Portainer?.ResourceControl}
+            resourceType={ResourceControlType.Network}
+            disableOwnershipChange={isSystemNetwork(networkQuery.data.Name)}
+            resourceId={networkId}
+          />
+        </>
+      )}
+      {networkQuery.data?.Options &&
+        Object.keys(networkQuery.data.Options).length > 0 && (
```

Comment:
we're doing this inside NetworkOptionsTable anyway (`Object.entries`). so might be more optimal to it in one go. I mean, remove the check here and in the component do something like:
```
function Component() {
	const entries = Object.entries(options);

  if (entries.length === 0 ) {
     return null
  }

  return <Widget>....
}

```

Reply:
updated